synapse/models/inet.py

```diff
@@ -715,43 +716,85 @@ def _normPyStr(self, valu):
         hostparts = ''
         pathpart = ''
         parampart = ''
+        isUNC = False
+        prep = False
 
         # Protocol
-        try:
-            proto, valu = valu.split('://', 1)
-            proto = proto.lower()
-            subs['proto'] = proto
-        except Exception:
+        for splitter in ('://///', ':////'):
+            try:
+                proto, valu = orig.split(splitter, 1)
+                proto = proto.lower()
+                assert proto == 'file'
+                isUNC = True
+                break
+            except Exception:
+                proto = valu = ''
+
+        if not proto:
+            try:
+                proto, valu = orig.split('://', 1)
+                proto = proto.lower()
+            except Exception:
+                pass
+
+        if not proto:
+            try:
+                proto, valu = orig.split(':/', 1)
+                proto = proto.lower()
+                assert proto == 'file'
+                prep = True
+            except Exception:
+                proto = valu = ''
+
+        if not proto:
+            try:
+                proto, valu = orig.split(':', 1)
+                proto = proto.lower()
+                assert proto == 'file'
+                assert valu
+                prep = True
+            except Exception:
+                proto = valu = ''
+
+        if not proto or not valu:
             raise s_exc.BadTypeValu(valu=orig, name=self.name,
                                     mesg='Invalid/Missing protocol') from None
 
+        subs['proto'] = proto
         # Query params first
         queryrem = ''
         if '?' in valu:
             valu, queryrem = valu.split('?', 1)
             # TODO break out query params separately
 
+        # Optional User/Password
+        parts = valu.rsplit('@', 1)
+        if len(parts) == 2:
+            authparts, valu = parts
+            prep = False
+            userpass = authparts.split(':', 1)
+            subs['user'] = userpass[0]
+            if len(userpass) == 2:
+                subs['passwd'] = userpass[1]
+
         # Resource Path
+        if prep:
+            valu = f'/{valu}'
         parts = valu.split('/', 1)
+        subs['path'] = ''
         if len(parts) == 2:
             valu, pathpart = parts
-            pathpart = f'/{pathpart}'
-        subs['path'] = pathpart
+            if proto == 'file' and windre.match(pathpart):
```

Comment:
A inline comment here to help future self may be a nice thing to have since it could be confusing to see the two same lines but inverted.

Reply:
:+1: